charts/metal3-deploy/0.1.0/charts/mariadb/templates/secret.yaml

```diff
@@ -1,9 +1,20 @@
+{{- $secret_name := "ironic-mariadb" -}}
+
 apiVersion: v1
 kind: Secret
 metadata:
-  name: ironic-mariadb
+  name: {{ $secret_name }}
   labels:
     {{- include "mariadb.labels" . | nindent 4 }}
-type: Opaque    
+type: Opaque
 data:
-  password: {{ .Values.password | toString | b64enc }}
+  # try to get the old secret
+  {{- $old_sec := lookup "v1" "Secret" .Release.Namespace $secret_name }}
+  # check, if a secret is already set
+  {{- if or (not $old_sec) (not $old_sec.data) }}
+  # if not set, then generate a new password
+  password: {{ randAlphaNum 20 | b64enc }}
```

Comment:
I think we should probably still allow a value to be set via the values file here e.g:

- If an existing secret exists, use that
- Otherwise check if there is a values file password specified (e.g default to an empty string instead of `change`)
- Else we generate a random password

This also has the advantage of working on upgrade for any users who did generate their own values files with something other than `change` 😄